doc/sfp-cmis/cmis-init.md

```diff
@@ -0,0 +1,216 @@
+should display the advertised applications of the transceiver.
+- If the transceiver is a CMIS module, the **sfputil** should be capable of detecting the errors of the CMIS application initialization.
+
+# Design
+
+## sonic-platform-daemons/sonic-xcvrd (modified)
+
+The transceiver daemon will be enhanced as below:
+
+- Post the CMIS application advertisement into the **STATE_DB**  
+  In the case of CMIS transceiver, the **application_advertisement** of **TRANSCEIVER_INFO** is as follow:  
+
+```
+  "TRANSCEIVER_INFO|Ethernet0": {
+    "type": "hash",
+    "value": {
+      "application_advertisement": "{1: {'ap_sel_code_id': 1, 'host_electrical_interface_id': '400GAUI-8 C2M (Annex 120E)', 'module_media_interface_id': '400GBASE-DR4 (Cl 124)', 'host_lane_count': 8, 'media_lane_count': 4, 'host_lane_assignment_options': 1, 'media_lane_assignment_options': None}, 2: {'ap_sel_code_id': 2, 'host_electrical_interface_id': '100GAUI-2 C2M (Annex 135G)', 'module_media_interface_id': '100G-FR/100GBASE-FR1 (Cl 140)', 'host_lane_count': 2, 'media_lane_count': 1, 'host_lane_assignment_options': 85, 'media_lane_assignment_options': None}}",
+      ...... omitted ......
+    }
+  },
+  ...... omitted ......
+```  
+
+- Add a CMIS manager class to handle the application initializations in parallel  
+
+  ![](images/001.png)
+
+## sonic-platform-common/sonic_platform_base/sfp_base.py (modified)
+
+Add the following stub routines for the CMIS transceiver
+- get_cmis_application_selected(self)  
+Retrieves the selected application code of this CMIS transceiver
+- get_cmis_application_matched(self, host_speed, host_lanes)  
+Retrieves the matched application code by the host interface config
+- set_cmis_application(self, host_speed, host_lanes)  
+Updates the application selection of this CMIS transceiver
+
+## sonic-platform-common/sonic_platform_base/sonic_xcvr/api/public/cmis.py (modified)
+
+- Add support for software reset
+- Add support for activating the CMIS application base on the host port mode.
+
+## sonic-platform-common/sonic_platform_base/sonic_xcvr/fields/consts.py (modified)
+
+- Add register definitions for CMIS application initialization
+
+## sonic-platform-common/sonic_platform_base/sonic_xcvr/fields/public/cmis.py (modified)
+
+- Add support for parsing CMIS application advertisement
+
+## sonic-platform-common/sonic_platform_base/sonic_xcvr/mem_maps/public/cmis.py (modified)
+
+- Add the meory map for the CMIS registers which is necessary for application initialization
+
+## sonic-platform-common/sonic_platform_base/sonic_xcvr/sfp_optoe_base.py (modified)
+
+- Add support for CMIS application initialization
+
+## sonic-utilities/sfputil (modified)
+
+- Add supprot to show the CMIS application advertisement
+
+## CLI commands
+
+### CLI Show Commands
+
+With CMIS application advertisement support, **show interfaces transceiver eeprom** is now as below.
+
+```
+admin@sonic:~$ show interfaces transceiver eeprom Ethernet0
```

Comment:
current show command output is like this:

```
admin@sonic:~$ show interface transceiver eeprom Ethernet0
Ethernet0: SFP EEPROM detected
        Application Advertisement: 400G CR8 - Copper cable
                                   200GBASE-CR4 (Clause 136) - Copper cable
                                   100GBASE-CR2 (Clause 136) - Copper cable
                                   100GBASE-CR4 (Clause 92) - Copper cable
                                   50GBASE-CR (Clause 126) - Copper cable
                                   50GBASE-CR2 with no FEC - Copper cable
                                   25GBASE-CR CA-N (Clause 110) - Copper cable
                                   1000BASE -CX(Clause 39) - Copper cable
                                   
        Connector: No separable connector
        Encoding: Not supported for CMIS cables
        Extended Identifier: Power Class 1(0.25W Max)
        Extended RateSelect Compliance: Not supported for CMIS cables
        Identifier: QSFP-DD Double Density 8X Pluggable Transceiver
        Length Cable Assembly(m): 0.5
        Nominal Bit Rate(100Mbs): Not supported for CMIS cables
        Specification compliance: Not supported for CMIS cables
        Vendor Date Code(YYYY-MM-DD Lot): 2021-05-14 
        Vendor Name: Mellanox
        Vendor OUI: 00-02-c9
        Vendor PN: MCP1660-W00AE30
        Vendor Rev: A3
        Vendor SN: MT2120VS03875
```

Reply:
Yes, the old one directly output the raw text fetched from the DB